charts/fleet/tests/controller-replica-count.yaml

```
@@ -0,0 +1,20 @@
+suite: replica values test for fleet controller
+templates:
+  - deployment.yaml
+  - deployment_gitjob.yaml
+  - deployment_helmops.yaml
+tests:
+  - it: should set spec.replicas to the number of replicas specified in agent.replicas
```

Is this the result of a copy/paste operation?
How about something like this, or ideally a bit shorter with a similar idea:
```suggestion
  - it: should set spec.replicas for each controller to the number of replicas specified in the controller's replicas field
```

Also, should we set the same value for all deployments in this test? This wouldn't detect issues such as, say, the gitOps controller deployment using the replica count of helmOps.

---

I have expanded the tests into three files so that each controller gets only the value set that it should respect, thereby testing specifically for values that are only related to the deployment manifest being tested.

Note that it is not possible to have those three tests in a single file if they have different values to be tested. This was only possible as long they have had the same values passed to it.